web/modules/custom/weather_block/src/Services/FetchApiData.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Drupal\weather_block\Services;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Http\ClientFactory;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use GuzzleHttp\Exception\ClientException;
+
+/**
+ * Fetch a data from api by city name.
+ */
+class FetchApiData {
+
+  /**
+   * Initialize a database property.
+   */
+  public function __construct(protected Connection $database, protected LoggerChannelFactoryInterface $logger, protected ClientFactory $httpClient) {
+  }
+
+  /**
+   * Private function for getting array with temperature.
+   */
+  public function getDataFromApi(string $cities, string $api_key): array {
```

Comment:
Api key is always the same, I think then instead of getting this key every time you want to use that method, I think it is better to let the service handle that.